### ISSUE TITLE
MS Teams bad request to graph api

### ISSUE THREAD

**Issue by EclipseKnight** [calcom/cal.com#2818]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

MS Teams api throws bad request when making a post request to the graph api on createMeeting function call in the VideoApiAdapter.ts file.

```
@calcom/web:start: 19:37:14.658 timeZoneName DEBUG [api] book:user Booking eventType 22 started
@calcom/web:start: calendarBusyTimes==>>> [
@calcom/web:start:   {
@calcom/web:start:     start: '2022-05-20T12:00:00.0000000Z',
@calcom/web:start:     end: '2022-05-20T20:30:00.0000000Z'
@calcom/web:start:   }
@calcom/web:start: ]
@calcom/web:start: 19:37:15.363 timeZoneName ERROR [lib] videoClient createMeeting failed
@calcom/web:start:  Error  Bad Request
@calcom/web:start: error stack:
@calcom/web:start: • 6341.js:1127 handleErrorsJson
@calcom/web:start:     .next/server/chunks/6341.js:1127:15
@calcom/web:start:
@calcom/web:start: • 8595.js:2273 refreshAccessToken
@calcom/web:start:     .next/server/chunks/8595.js:2273:71
@calcom/web:start:
@calcom/web:start: • 8595.js:2330 createMeeting
@calcom/web:start:     .next/server/chunks/8595.js:2330:33
@calcom/web:start:
@calcom/web:start: • 8595.js:7875 createMeeting
@calcom/web:start:     .next/server/chunks/8595.js:7875:28
@calcom/web:start:
@calcom/web:start: • 8595.js:7199 create
@calcom/web:start:     .next/server/chunks/8595.js:7199:28
@calcom/web:start:
@calcom/web:start: • event.js:1510 handler
@calcom/web:start:     .next/server/pages/api/book/event.js:1510:31
@calcom/web:start:
@calcom/web:start: • node.ts:248 apiResolver
@calcom/web:start:     node_modules/next/server/api-utils/node.ts:248:5
@calcom/web:start:
@calcom/web:start: • next-server.ts:551 runApi
@calcom/web:start:     node_modules/next/server/next-server.ts:551:5
@calcom/web:start:
@calcom/web:start: • base-server.ts:810 fn
@calcom/web:start:     node_modules/next/server/base-server.ts:810:27
@calcom/web:start:
@calcom/web:start: • router.ts:413 execute
@calcom/web:start:     node_modules/next/server/router.ts:413:24
@calcom/web:start:
@calcom/web:start: • base-server.ts:963 run
@calcom/web:start:     node_modules/next/server/base-server.ts:963:23
@calcom/web:start:
@calcom/web:start: • base-server.ts:576 handleRequest
@calcom/web:start:     node_modules/next/server/base-server.ts:576:14
@calcom/web:start:
@calcom/web:start:
@calcom/web:start: {
@calcom/web:start:   type: 'Test',
@calcom/web:start:   title: 'Test between organizer and test',
@calcom/web:start:   description: '',
@calcom/web:start:   additionalNotes: 'jhjhkjhkjh\n',
@calcom/web:start:   startTime: '2022-05-20T16:30:00-04:00',
@calcom/web:start:   endTime: '2022-05-20T16:45:00-04:00',
@calcom/web:start:   organizer: {
@calcom/web:start:     name: 'Jon Paquin',
@calcom/web:start:     email: 'organizer@example.com',
@calcom/web:start:     timeZone: 'America/Detroit',
@calcom/web:start:     language: {
@calcom/web:start:       translate: [Function: fixedT] {
@calcom/web:start:         lng: 'en',
@calcom/web:start:         ns: 'common',
@calcom/web:start:         keyPrefix: undefined
@calcom/web:start:       },
@calcom/web:start:       locale: 'en'
@calcom/web:start:     }
@calcom/web:start:   },
@calcom/web:start:   attendees: [
@calcom/web:start:     {
@calcom/web:start:       email: 'attendee@example.com',
@calcom/web:start:       name: 'test',
@calcom/web:start:       timeZone: 'America/New_York',
@calcom/web:start:       language: {
@calcom/web:start:         translate: [Function: fixedT] {
@calcom/web:start:           lng: 'en',
@calcom/web:start:           ns: 'common',
@calcom/web:start:           keyPrefix: undefined
@calcom/web:start:         },
@calcom/web:start:         locale: 'en'
@calcom/web:start:       }
@calcom/web:start:     }
@calcom/web:start:   ],
@calcom/web:start:   location: 'integrations:office365_video',
@calcom/web:start:   destinationCalendar: {
@calcom/web:start:     id: 8,
@calcom/web:start:     integration: 'office365_calendar',
@calcom/web:start:     externalId: 'AQMkADQ1OTRiZGVlAC1hZDdlLTQ3OGItODMwYS1mZWFjYjVlMzkyMWEARgAAA-T33UTevJVMuZJkREuhL7MHAM2UF7v_bR5IiVavhzqIK1QAAAIBBgAAAM2UF7v_bR5IiVavhzqIK1QAAAIvBAAAAA==',
@calcom/web:start:     userId: 9,
@calcom/web:start:     bookingId: 14,
@calcom/web:start:     eventTypeId: null
@calcom/web:start:   },
@calcom/web:start:   hideCalendarNotes: false,
@calcom/web:start:   uid: 'pwG2mFagRxhAUEVzPFFAv9',
@calcom/web:start:   conferenceData: {
@calcom/web:start:     createRequest: {
@calcom/web:start:       requestId: '928381d5-f372-5646-9145-0aa351854108'
@calcom/web:start:     }
@calcom/web:start:   }
@calcom/web:start: }
@calcom/web:start: {
@calcom/web:start:   error: 'invalid_grant',
@calcom/web:start:   error_description: 'AADSTS9002313: Invalid request. Request is malformed or invalid.\r\n' +
@calcom/web:start:     'Trace ID: f5089632-3f7f-4310-8481-0b9784a00d00\r\n' +
@calcom/web:start:     'Correlation ID: 76c5c84d-e8a8-4671-a688-6c6284fdc423\r\n' +
@calcom/web:start:     'Timestamp: 2022-05-18 19:37:15Z',
@calcom/web:start:   error_codes: [ 9002313 ],
@calcom/web:start:   timestamp: '2022-05-18 19:37:15Z',
@calcom/web:start:   trace_id: 'f5089632-3f7f-4310-8481-0b9784a00d00',
@calcom/web:start:   correlation_id: '76c5c84d-e8a8-4671-a688-6c6284fdc423',
@calcom/web:start:   error_uri: 'https://login.microsoftonline.com/error?code=9002313'
@calcom/web:start: }
```

### Steps to Reproduce

1. clone clean repository following guide in readme
2. configure Office environment variables appropriately
3. try to schedule email with event-type location office365_video

It sends 1 unformatted (no html) receipt like email and the formatted html confirmation email. 
Neither have a teams meeting url (and dial in info which would be useful).

I've gone through the api code thoroughly and have found a couple points of concern so feel to ask me any further questions that I may have missed.

**Comment by EclipseKnight** [comment]
In addition, it seems the issue that also arises is an undefined refresh token being passed. Added some print outs to verify this 
```
[32m@calcom/web:start: [39mgetAvailability() function called...
[32m@calcom/web:start: [39mcalendarBusyTimes==>>> []
[32m@calcom/web:start: [39mcreateMeeting() function called...
[32m@calcom/web:start: [39mrefreshAccessToken() function called...
[32m@calcom/web:start: [39m
[32m@calcom/web:start: [39m    Request POST: https://login.microsoftonline.com/common/oauth2/v2.0/token
[32m@calcom/web:start: [39m
[32m@calcom/web:start: [39m    client_id: fakeclientidforexample
[32m@calcom/web:start: [39m    refresh_token: undefined;
[32m@calcom/web:start: [39m    client_secret: fakesecretforexample
[32m@calcom/web:start: [39m    
[32m@calcom/web:start: [39mResponse: [object Response]
```

**Comment by scantreed** [comment]
I'm running into the same issue when I try to set up the o365 calendar app for the first time

**Comment by scantreed** [comment]
I believe this problem is occurring because the env values are not being properly passed to the submodules and the if statement on https://github.com/calcom/cal.com/blob/498759890f4b3576ec77cb1cd0080a92843ae631/packages/prisma/seed-app-store.ts#L48 is failing because it thinks the client and graph ID are set to NULL

 I was able to temporarily fix this issue by changing the if statement to if(true) and hardcoding the client id and secret for the outlook integration

**Comment by EclipseKnight** [comment]
@scantreed Nice find! I was looking through everything but it happens that was the one condition I didn't think to check. I'll give that a shot and see if I can also troubleshoot where the null value is coming from.

On a side note, do you know if it actually provides the teams meeting url and or video conferencing data once working? I'll be working to add those as well if not.

**Comment by EclipseKnight** [comment]
Actually, I may have spoken a bit too soon because in one of my logs I printed out the client_id and client_secret being passed to the request, and both were appropriately set and passed along with the POST requests. So in my case, I don't think this is the cause of the issue. I'll still give your workaround a shot to maybe see if there is something else that could be missing and update here with the results.

**Comment by scantreed** [comment]
Thanks @EclipseKnight 

It's interesting to hear that your environment variables may be being correctly passed. If that is the case it I'd be curious to hear more about how you set up your environment. 

Regarding the data once it's actually working, I'm not using the teams integration, I'm actually using the outlook integration. I haven't been able to fully verify that it works with my AD tentant (because I need to associate a cert to the webserver hosting it as AD only accepts encrypted traffic), but it does successfully redirect me to the AD tenant, so it seems to be working. I'll provide a further update when I get the chance to test under an encrypted connection.

**Comment by hariprasadiit** [comment]
I've found where the issue is for this.

When you add only MS Teams app without Office 365 Calendar app, for initial request to get `access_token` only `OnlineMeetings.ReadWrite` scope is used
but for subsequent requests to get refresh `access_token` `"User.Read", "OnlineMeetings.ReadWrite"` scopes are used for which MS Graph responds with `invalid_grant` error as scopes are not matching.

This is will not be there if Office 365 Calendar app is added as it makes initial request with `"User.Read"` scope which will work for MS Teams

**Comment by PeerRich** [comment]
cc @joeauyeung maybe you can take a look? Or can this be due to the verification? 🤔

**Comment by PeerRich** [comment]
@joeauyeung was this fixed?

**Comment by joeauyeung** [comment]
Fixed in #4910
